Refine #256

terencode opened the pull request:
- ~~Disable 'Export HW port' when JACK is not running~~
- Fix 'Export HW port' tray option not working and incorrect state
- Disable 'Export HW port' when ports are already exported and notify it
with the button text
- Do not start A2JMIDI after exporting ports
- Stop A2JMIDI before stopping JACK


~~Reasoning:~~
- ~~It doesn't make sense to only export ports
while not starting A2J: Either you don't want to export the ports at all
if for example your soundcard crashes A2J when exporting its ports, or
you want your ports exported which means it starts A2J as well.~~
- ~~If you plug a new hw, just export hw ports again and it will stop/start
A2J if needed~~.

terencode commented:
@falkTX is my reasoning correct?

falkTX commented:
A button or other UI element should not do more than what it says, otherwise confuses the user.

Saving a config should not start a tool, as you might have misclicked or still have other things in mind to change before actually starting up the process.

terencode commented:
> A button or other UI element should not do more than what it says, otherwise confuses the user.

That's a good point.

Maybe I wasn't very clear but this is for the button 'Export HW Port' and not for the checkboxes, so it has nothing to do with the config.

terencode commented:
Also I just realised that once you export hw ports once, it seems it automatically detects new ones being connected, so if ports are already exported, it should disable the button.

terencode commented:
I changed the behaviour base on your feedback, fixed an unrelated issue where stopping JACK would not stop A2JMIDI before and made the button change text to notify the user ports are already exported.

falkTX commented:
Alright, merging then. thanks
